### ISSUE TITLE
[TreeView] parent/child selectionBehavior

### ISSUE THREAD

**Issue by nathanforce** [adobe/react-spectrum#6589]
### Provide a general summary of the feature here

I'm having a hard time implementing parent/child selection behavior in a TreeView.  Specifically I'd like to be able to: 

1. select a parent and all of its children by selecting the parent checkbox.
2. Unselecting a child while siblings are still selected switches the parent to indeterminate
3. Unselecting the last child unselects the parent
4. Should `onSelectionChange` include parent keys? In my case parents are basically just meant for grouping and in my external state I'm only interested in knowing which child items are selected.

### 🤔 Expected Behavior?

Support intuitive parent/child selection.

### 😯 Current Behavior

Parents and children feel unrelated

### 🔦 Context

I'm building this sort of CheckboxTree, which I suspect is a pretty common usage of Tree.

<img width="478" alt="image" src="https://github.com/adobe/react-spectrum/assets/6694194/f386b8d4-6236-4172-9654-fb91c3ba70a9">

**Comment by LFDanLu** [comment]
We'll be most likely looking to add this behavior to our implementation of RSP TreeView (and by extension RAC Tree) since it is supported by Spectrum Design, but I'm unsure when that work will be slated to be picked up. As for the implementation, I imagine [SelectionManager](https://github.com/adobe/react-spectrum/blob/4e3af379e569faac3b374e0ed5a98b2a19cd92c3/packages/%40react-stately/selection/src/SelectionManager.ts#L301) would need to be updated to have a mode where toggling the selection of a node that has children would also select the children. As for `onSelectionChange` including the parent keys I imagine we would still want to include them for use cases do care about parent key selection as well but maybe that could be changed via a option as well.

**Comment by imjhonny** [comment]
@nathanforce were you able to find an implementation to this? I am trying to do something similar as well.

**Comment by nathanforce** [comment]
Yes, but for now we've had to patch the package and manipulate the collection directly.

**Comment by raygesualdo** [comment]
Are there any plans to add this functionality? This is fairly common with `Tree` components from other UI libraries. For the moment, we've had to write a significant amount of custom code that handles this.

**Comment by LFDanLu** [comment]
It is certainly still something we'd like to support, but lower on our priorities unfortunately. We'd welcome a contribution for this, see https://github.com/adobe/react-spectrum/pull/8800#issuecomment-4070515732 for prior work and feedback if you are interested in picking this up.
